Fix saving media on Android using file:///

Menardi opened the pull request:
This is a fix to allow the saving of media from the application to the gallery.

The existing code seems to look for the file within the Android filesystem, rather than in the application. So, when I call cordova.plugins.photoLibrary.saveImage("file:///www/img/test.jpg"), it will look for a folder named www in the root of the filesystem, and so will never find the file I'm looking for.

Instead we need to look within the application's assets folder, which is where Cordova stores all of its files, which is what this PR does.

viskin commented:
I don't think this is correct thing to do.
If I want to save to file:///android_assets, your solution will work.
But what if I want to save video to **sdcard** directory (say file:///storage/emulated/0/)? Or **external cache directory** (file:///storage/emulated/0/Android/data/com.my.app/cache/")?
If you want to save to file:///www/img/test.jpg, why don't you use **cordova-plugin-file** and

      applicationDirectory + 'www/img/test.jpg'?

Current solution is cordova-plugin-file and you just provide a path using [one of these folders](https://github.com/apache/cordova-plugin-file#android-file-system-layout).

Menardi commented:
In this case I am not trying to write to assets, but read from it, and save an image from assets to the user's gallery. However, I didn't consider the use case of users wanting to move things from the cache to the gallery, which my PR would break. I have updated it to use the previous behaviour except when the url starts with file:///android_asset/.

The problem with using applicationDirectory is that it only returns "file:///android_asset/". This is not a unique reference to this app's asset folder (unlike it is on iOS). So, if you do _new File(new URI("file:///android_asset/www/img/test.jpg))_, it will look for a folder named android_asset on the root of the Android filesystem.

The only way to get access to the assets is by calling getAssets() on the context, which I've added in this PR.

viskin commented:
Ok, I think this is very good solution that @Menardi implemented - to check whether a path begins with "file:///android_asset/", and apply getAssets logic in such case.
I will merge this into master branch.

Menardi commented:
Glad the explanation made sense! Thanks for the quick response and merge.
